Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,271 @@
+  `L` is finite-dimensional.
+
+* `normal_iff_isGalois` : The fixing subgroup of an intermediate field `L` is normal if and only if
+  `L` is Galois.
+
+-/
+
+variable {k K : Type*} [Field k] [Field K] [Algebra k K]
+
+namespace InfiniteGalois
+
+open Pointwise FiniteGaloisIntermediateField AlgEquiv
+--Note: The `adjoin`s below are `FiniteGaloisIntermediateField.adjoin`
+
+lemma fixingSubgroup_isClosed (L : IntermediateField k K) [IsGalois k K] :
+    IsClosed (L.fixingSubgroup : Set (K ≃ₐ[k] K)) where
+    isOpen_compl := isOpen_iff_mem_nhds.mpr fun σ h => by
+      apply mem_nhds_iff.mpr
+      rcases Set.not_subset.mp ((mem_fixingSubgroup_iff (K ≃ₐ[k] K)).not.mp h) with ⟨y,yL,ne⟩
+      use σ • ((adjoin k {y}).1.fixingSubgroup : Set (K ≃ₐ[k] K))
+      constructor
+      · intro f hf
+        rcases (Set.mem_smul_set.mp hf) with ⟨g,hg,eq⟩
+        simp only [Set.mem_compl_iff, SetLike.mem_coe, ← eq]
+        apply (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).not.mpr
+        push_neg
+        use y
+        simp only [yL, smul_eq_mul, AlgEquiv.smul_def, AlgEquiv.mul_apply, ne_eq, true_and]
+        have : g y = y := (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp hg y <|
+          adjoin_simple_le_iff.mp fun ⦃x⦄ a ↦ a
+        simpa only [this, ne_eq, AlgEquiv.smul_def] using ne
+      · simp only [(IntermediateField.fixingSubgroup_isOpen (adjoin k {y}).1).smul σ, true_and]
+        use 1
+        simp only [SetLike.mem_coe, smul_eq_mul, mul_one, and_true, Subgroup.one_mem]
+
+lemma fixedField_fixingSubgroup (L : IntermediateField k K) [IsGalois k K] :
+    IntermediateField.fixedField L.fixingSubgroup = L := by
+  apply le_antisymm
+  · intro x hx
+    rw [IntermediateField.mem_fixedField_iff] at hx
+    have mem : x ∈ (adjoin L {x}).1 := subset_adjoin _ _ rfl
+    have : IntermediateField.fixedField (⊤ : Subgroup ((adjoin L {x}) ≃ₐ[L] (adjoin L {x}))) = ⊥ :=
+      (IsGalois.tfae.out 0 1).mp (by infer_instance)
+    have : ⟨x, mem⟩ ∈ (⊥ : IntermediateField L (adjoin L {x})) := by
+      rw [← this, IntermediateField.mem_fixedField_iff]
+      intro f _
+      rcases restrictNormalHom_surjective K f with ⟨σ,hσ⟩
+      apply Subtype.val_injective
+      rw [← hσ, restrictNormalHom_apply (adjoin L {x}).1 σ ⟨x, mem⟩]
+      have := hx ((IntermediateField.fixingSubgroupEquiv L).symm σ)
+      simpa only [SetLike.coe_mem, true_implies]
+    rcases IntermediateField.mem_bot.mp this with ⟨y, hy⟩
+    obtain ⟨rfl⟩ : y = x := congrArg Subtype.val hy
+    exact y.2
+  · exact (IntermediateField.le_iff_le L.fixingSubgroup L).mpr fun ⦃x⦄ a ↦ a
+
+lemma fixedField_bot [IsGalois k K] :
+    IntermediateField.fixedField (⊤ : Subgroup (K ≃ₐ[k] K)) = ⊥ := by
+  rw [← IntermediateField.fixingSubgroup_bot, fixedField_fixingSubgroup]
+
+lemma restrict_fixedField (H : Subgroup (K ≃ₐ[k] K)) (L : IntermediateField k K) [IsGalois k L] :
+    (IntermediateField.fixedField H) ⊓ L = IntermediateField.lift (IntermediateField.fixedField
```

Comment:
This would benefit from `open IntermediateField`.

Reply:
I think I alredy opened `FiniteGaloisIntermediateField`, opening it too may cause trouble on `adjoin`

Reply:
Opened, thanks.